src/common/lockdep.cc

```diff
@@ -52,11 +52,12 @@ static lockdep_stopper_t lockdep_stopper;
 static ceph::unordered_map<std::string, int> lock_ids;
 static map<int, std::string> lock_names;
 static map<int, int> lock_refs;
-static list<int> free_ids;
+static char free_ids[MAX_LOCKS/8]; // bit set = free
```

Comment:
For clarity it might be good, but then, we'll miss searching for a set bit by 8 bits at a time (https://github.com/ceph/ceph/pull/9254/commits/b3e24629556eb079b9ed8a3cdf626988b96a329a#diff-6bd21e92bb10c9be3d6c505cdefc142dR154) and (re)setting by at least 32 bits at a time (https://github.com/ceph/ceph/pull/9254/commits/b3e24629556eb079b9ed8a3cdf626988b96a329a#diff-6bd21e92bb10c9be3d6c505cdefc142dR83 and https://github.com/ceph/ceph/pull/9254/commits/b3e24629556eb079b9ed8a3cdf626988b96a329a#diff-6bd21e92bb10c9be3d6c505cdefc142dR108).